scipy/diff/numdifftools/tests/test__step_generators.py

```diff
@@ -0,0 +1,46 @@
+import numpy as np
+from numpy.testing import (assert_, run_module_suite)
+from scipy.diff.numdifftools._step_generators import _generate_step
+
+
+def steps(base_step, step_ratio, num_steps, sign):
+    for i in range(num_steps):
+            steps = base_step * step_ratio ** (sign*i)
+            if (np.abs(steps) > 0).all():
+                yield steps
+
+
+class Test(object):
+    def test_min_step(self):
```

Comment:
The basic rationale is that step generators generates a geometric progression. So, the prediction of answer will involve the exact code as there is already in step_generators.